main.go

```diff
@@ -83,42 +83,56 @@ func main() {
 		log.Log.Infow("Initialized connection to Redis", "i", i)
 
 		aggregatedRedisClients[i] = redisClients
+		clientsSize := len(redisClients)
+
+		redisPubsAggregationEntry := make([]chan<- *redispub.Publication, clientsSize)
+		stopRedisPubsEntry := make([]chan bool, clientsSize)
+
+		for j := 0; j < clientsSize; j++ {
```

Comment:
general comment, this has grown somewhat hard to follow because of the nested arrays and loops, etc. it would be nice to refactor it to be a bit more abstract when we're back on firmer ground.

Reply:
I think type aliases will help with this somewhat, but will also add some comments.